pkg/resource/podtemplatespec_test.go

```diff
@@ -235,3 +240,101 @@ func verifyMount(mount corev1.VolumeMount, expected *volumeMount, t *testing.T)
 		t.Errorf("expected mount path to be %s, got %s", expected.mountPath, mount.MountPath)
 	}
 }
+
+func TestMakePodTemplateSpecS3CloudFront(t *testing.T) {
+	ctx := context.Background()
+
+	testBuilder := cirofake.NewFixturesBuilder()
+	config := &v1.Config{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Spec: v1.ImageRegistrySpec{
+			Storage: v1.ImageRegistryConfigStorage{
+				ManagementState: "Unmanaged",
+				S3: &v1.ImageRegistryConfigStorageS3{
+					Bucket:  "bucket",
+					Region:  "region",
+					Encrypt: true,
+					CloudFront: &v1.ImageRegistryConfigStorageS3CloudFront{
+						BaseURL:   "https://cloudfront.example.com",
+						KeypairID: "keypair-id",
+						Duration: metav1.Duration{
+							Duration: 300 * time.Second,
+						},
+					},
+					VirtualHostedStyle: true,
+				},
+			},
+		},
+	}
+	testBuilder.AddRegistryOperatorConfig(config)
+
+	infra := &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Status: configv1.InfrastructureStatus{
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: configv1.AWSPlatformType,
+				AWS: &configv1.AWSPlatformStatus{
+					Region: "region",
+				},
+			},
+		},
+	}
+	testBuilder.AddInfraConfig(infra)
+
+	imageRegNs := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "openshift-image-registry",
+			Annotations: map[string]string{
+				"openshift.io/sa.scc.supplemental-groups": "1000430000/10000",
+			},
+		},
+	}
+	testBuilder.AddNamespaces(imageRegNs)
+
+	fixture := testBuilder.Build()
+	s3Storage := s3.NewDriver(ctx, config.Spec.Storage.S3, fixture.Listers)
+	pod, _, err := makePodTemplateSpec(fixture.KubeClient.CoreV1(), fixture.Listers.ProxyConfigs, s3Storage, config)
+	if err != nil {
+		t.Fatalf("error creating pod template: %v", err)
+	}
+
+	ignoreEnvVar := func(name string) bool {
+		return !strings.HasPrefix(name, "REGISTRY_STORAGE") && !strings.HasPrefix(name, "REGISTRY_MIDDLEWARE")
+	}
+	expectedEnvVars := map[string]corev1.EnvVar{
+		"REGISTRY_STORAGE":                                  {Value: "s3"},
+		"REGISTRY_STORAGE_S3_BUCKET":                        {Value: "bucket"},
+		"REGISTRY_STORAGE_S3_REGION":                        {Value: "region"},
+		"REGISTRY_STORAGE_S3_ENCRYPT":                       {Value: "true"},
+		"REGISTRY_STORAGE_S3_VIRTUALHOSTEDSTYLE":            {Value: "true"},
+		"REGISTRY_STORAGE_S3_USEDUALSTACK":                  {Value: "true"},
+		"REGISTRY_STORAGE_S3_CREDENTIALSCONFIGPATH":         {Value: "/var/run/secrets/cloud/credentials"},
+		"REGISTRY_MIDDLEWARE_STORAGE_CLOUDFRONT_BASEURL":    {Value: "https://cloudfront.example.com"},
```

Comment:
I mentioned it in the BZ in case it is helpful, but https://stackoverflow.com/a/69089002/6431503 explained the only environment variable structure that worked for me while trying to debug this problem.